### PR TITLE
VideoPlayer: Defer before the next frame the default thumbnail and annotations loading

### PR DESCRIPTION
**Pull request by misl6**
Fixes issue #7755 

Maintainer merge checklist
* [ ] Title is descriptive/clear for inclusion in release notes.
* [ ] Applied a `Component: xxx` label.
* [ ] Applied the `api-deprecation` or `api-break` label.
* [ ] Applied the `release-highlight` label to be highlighted in release notes.
* [ ] Added to the milestone version it was merged into.
* [ ] **Unittests** are included in PR.
* [ ] Properly documented, including `versionadded`, `versionchanged` as needed.

### CODE REVIEW

File: kivy/uix/videoplayer.py [hunk 1]
```diff
@@ -501,47 +499,58 @@ def _trigger_video_load(self, *largs):
...
+        Clock.schedule_once(self._try_load_default_annotations, -1)
         if self._video is not None:
             self._video.unload()
             self._video = None
         if value:
             self._trigger_video_load()
 
+    def on_thumbnail(self, instance, value):
+        if value:
+            self._load_thumbnail(value)
```
Comment (matham) [inline kivy/uix/videoplayer.py:529]: Do we need to reset to default if value is None? Or maybe clear it?

**Review by matham — CHANGES_REQUESTED** [review]

File: kivy/uix/videoplayer.py [hunk 2]
```diff
@@ -501,47 +499,58 @@ def _trigger_video_load(self, *largs):
...
+        # with the same filename (except extension) of the source (video) file.
+        Clock.schedule_once(self._try_load_default_thumbnail, -1)
+        Clock.schedule_once(self._try_load_default_annotations, -1)
         if self._video is not None:
             self._video.unload()
             self._video = None
         if value:
             self._trigger_video_load()
 
+    def on_thumbnail(self, instance, value):
```
[the remark this answers is not in the document]
Comment (misl6) [reply kivy/uix/videoplayer.py]: IIRC: On docs, we have a warning regarding `on_xxx` inheritance
`fbind` looks a great approach for things inside the framework BTW, applied the fix.

File: kivy/uix/videoplayer.py (hunk 1 above)
Comment (misl6) [reply kivy/uix/videoplayer.py]: Now the user can clear it. Added a test that checks it.

File: kivy/uix/videoplayer.py [hunk 3]
```diff
@@ -501,47 +504,57 @@ def _trigger_video_load(self, *largs):
...
             return
         self._annotations_labels = []
-        annotations = self.annotations
-        if not annotations:
-            filename = self.source.rsplit('.', 1)
-            annotations = filename[0] + '.jsa'
-        if exists(annotations):
-            with open(annotations, 'r') as fd:
-                self._annotations = load(fd)
+        with open(annotations, 'r') as fd:
```
Comment (matham) [inline kivy/uix/videoplayer.py:556]: Couldn't `annotations` now be the empty string? If so we also need to check if it's empty!?

Comment (misl6) [reply kivy/uix/videoplayer.py]: 🤦 You're right. Added the check and a test.

**Review by matham — DISMISSED** [review]

**Pull request merged**